### PR TITLE
feat(sdk): url path variables (simulator)

### PR DESCRIPTION
**Pull request by tsuf239**
fixes https://github.com/winglang/wing/issues/1851

added: 
- path vars support to the simulator
- regex validation for the [api supported URLs](https://regexr.com/7bopt)

tests: 
- retrieving the req.vars and sending them back in the response body
- testing a few valid and invalid urls

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by eladb** [comment]
updated title (the title should be easy to understand from a user perspective - think about the changelog)

File: examples/tests/valid/api_path_vars.w [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+let handler = inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
+    let vars = req.vars ?? {name: ""};
+    return cloud.ApiResponse {
+    body: {user: vars.get("name")},
+    status: 200
+  };
+ 
+};
```
Comment (eladb) [inline examples/tests/valid/api_path_vars.w:17]: Indentation looks weird

File: examples/tests/valid/api_path_vars.w [hunk 2]
```diff
@@ -0,0 +1,32 @@
+bring cloud;
+
+let api = new cloud.Api();
+
+resource Fetch {
+  extern "./api_path_vars.js" inflight get(url:str): Json;
+  init(){}
+}
```
Comment (eladb) [inline examples/tests/valid/api_path_vars.w:8]: Let's try to keep our Wing code well formatted to set a good example for the community:

```suggestion
resource Fetch {
  extern "./api_path_vars.js" inflight get(url: str): Json;
  init() { }
}
```

File: examples/tests/valid/api_path_vars.w [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  };
+ 
+};
+
+api.get("/users/{name}", handler);
+
+let f = new Fetch();
+
+new cloud.Function(inflight () => {
+let username = "tsuf";
```
Comment (eladb) [inline examples/tests/valid/api_path_vars.w:24]: indentation

File: examples/tests/valid/api_valid_path.w [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+try {
+api.get("/test/path/{with}/{two:invali4d#}/variables", handler);
+} catch error {
+  assert(error == "Invalid route /test/path/{with}/{two:invali4d#}/variables. Routes and params should consist of alpha-numeric characters only.");
+}
+
+try {
+api.get("/test/path/{unclosed", handler);
```
Comment (eladb) [inline examples/tests/valid/api_valid_path.w:39]: Indentation all around

File: examples/tests/valid/api_path_vars.w [hunk 5]
```diff
@@ -0,0 +1,32 @@
+bring cloud;
```
[the remark this answers is not in the document]
Comment (tsuf239) [reply examples/tests/valid/api_path_vars.w:1]: sure :)

File: examples/tests/valid/package.json [hunk 6]
```diff
@@ -2,6 +2,7 @@
   "dependencies": {
     "@cdktf/provider-aws": "12.0.5",
     "jsii-code-samples": "1.7.0",
+    "node-fetch": "^3.3.1",
```
[the remark this answers is not in the document]
Comment (tsuf239) [reply examples/tests/valid/package.json:5]: I thought that too! but it seems like the "extern" js code doesn't...

File: examples/tests/valid/api_valid_path.w (hunk 4 above)
Comment (tsuf239) [reply examples/tests/valid/api_valid_path.w]: Thanks!! fixed for the two .w files

File: examples/tests/valid/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (eladb) [reply examples/tests/valid/package.json:5]: I think that's due to the fact that we are not propagating it into the VM that runs the inflights. https://github.com/winglang/wing/blob/60520b9c4400beebe2c35e34fccf872f471f0026/libs/wingsdk/src/target-sim/function.inflight.ts#L49

[the remark this answers is not in the document]
Comment (Chriscbr) [reply examples/tests/valid/package.json:5]: @staycoolcall911 sounds good to me, I think it's ok adding `node-fetch` as a helper in the examples as a temporary workaround

**Comment by Chriscbr** [comment]
@tsuf239 When you're ready for another review, please click the "re-request review" button next to each reviewer on the top-right section of the pull request page

**Review by Chriscbr — APPROVED** [review]
Great work

**Pull request merged**